### PR TITLE
[CWS] do not resolve new fields in functional test field collector

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

When collecting fields values in the functional tests event collector, we might end up resolving fields that were not resolved during regular evaluation which is sub-optimal and could hide bugs (for example hashing files that were not hashed in the evaluation).

This PR tracks the fields that are resolved, and uses this to skip the resolution of fields that have not been evaluated.

This is a complete no-op outside of functional tests.

this reverts https://github.com/DataDog/datadog-agent/pull/31268

### Describe how to test/QA your changes

This is test debugging infrastructure.

### CODE REVIEW

**Comment by YoannGh** [comment]
Your PR should probably revert #31268 entirely

File: pkg/security/secl/compiler/eval/context_funtests.go
Comment (spikat) [inline pkg/security/secl/compiler/eval/context_funtests.go:1]: Should it be _functests.go instead of funtests.go?

Comment (paulcacheux) [reply pkg/security/secl/compiler/eval/context_funtests.go:1]: absolutely !

**Review by spikat — APPROVED** [review]

**Review by safchain — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Pull request merged**